obp/ope/estimators_slate.py

```diff
@@ -566,6 +567,291 @@ def estimate_interval(
         )
 
 
+@dataclass
+class SlateCascadeDoublyRobust(BaseSlateOffPolicyEstimator):
+    """Cascade Doubly Robust (Cascade-DR) Estimator.
+
+    Note
+    -------
+    Slate Cascade Doubly Robust (Cascade-DR) estimates the policy value of evaluation policy :math:`\\pi_e`
+    assuming the cascade click model (users interact with actions from the top position to the bottom in a slate).
+    It also uses reward prediction :math:`\\hat{Q}_k` as a control variate, which is derived using `obp.ope.SlateRegressionModel`.
+    Please refer to Section 3.1 of Kiyohara et al.(2022) for the detail.
+
+    Parameters
+    ----------
+    len_list: int
+        Length of a list of actions recommended in each impression (slate size).
+        When Open Bandit Dataset is used, 3 should be set.
+
+    n_unique_action: int
+        Number of unique actions.
+
+    estimator_name: str, default='cascade-dr'.
+        Name of the estimator.
+
+    References
+    ------------
+    Haruka Kiyohara, Yuta Saito, Tatsuya Matsuhiro, Yusuke Narita, Nobuyuki Shimizu, and Yasuo Yamamoto.
+    "Doubly Robust Off-Policy Evaluation for Ranking Policies under the Cascade Behavior Model.", 2022.
+
+    """
+
+    len_list: int
+    n_unique_action: int
+    estimator_name: str = "cascade-dr"
+
+    def __post_init__(self):
+        """Initialize Class."""
+        if self.n_unique_action is None:
+            raise ValueError("n_unique_action must be given")
+
+    def _estimate_round_rewards(
+        self,
+        action: np.ndarray,
+        reward: np.ndarray,
+        position: np.ndarray,
+        behavior_policy_pscore: np.ndarray,
+        evaluation_policy_pscore: np.ndarray,
+        q_hat: np.ndarray,
+        evaluation_policy_action_dist: np.ndarray,
+        **kwargs,
+    ) -> np.ndarray:
+        """Estimate rewards given round (slate_id) and slot (position).
+
+        Parameters
+        ----------
+        action: array-like, (<= n_rounds * len_list,)
```

Comment:
[ask]
When we use a Cascade-DR estimator, is the array shape equal to `n_rounds * len_list`? (not `<=`, but `==`?)

Reply:
Yes, the shape is equal to `n_rounds * len_list`.   
I just wonder, in which case the shape can be `<=` in IPS-based estimators?  

I try to be consistent with IPS-based estimators in this part, but maybe I don't fully understand the shape.
https://github.com/aiueola/zr-obp/blob/9543635aa137ba9b2ce31a1ece00d7d957f552de/obp/ope/estimators_slate.py#L64

Reply:
> I just wonder, in which case the shape can be `<= in` IPS-based estimators?

There would be a case where items of low positions are not exposed to the user (e.g., few items matched the query context).

IPS-based estimators seems to be implemented to handle such a case.

Reply:
I see.. I will fix the description in Cascade-DR from `<=` to `==`.